### DISCUSSION TITLE
What can we do when using html that comes from an api?

### DISCUSSION THREAD

**Discussion by Der-Alex** [tailwindlabs/tailwindcss#12382] [Help] [answered]
We have a project that uses an api as an endpoint for cms content and nuxt 3 with tailwind for the frontend. The api sends html snippets that can be created in an external backend. 
When I fetch some content from the api and use `<div class="content-wrapper" v-html="data" />`, nothing gets styled. I guess it has something to do with this: https://tailwindcss.com/docs/content-configuration#dynamic-class-names. 
There should be a solution for this use-case. 

So what can we do to make this work? 

My current solution is very hacky: I create a div inside the app.vue and comment it out: `<!-- <div class="all-my-tailwind-classes-i-need"></div>-->`

**Comment by Der-Alex** [comment]
Okay after searching through other discussions that were not really related to my use-case, I found that there is a configuration: https://tailwindcss.com/docs/content-configuration#safelisting-classes
This would definitely help me.

**Comment by alex-krasikau** [comment, marked as answer]
Hey @Der-Alex!

You need to find a way to feed HTML content that has classes to Tailwind.

1. If you get the content at build time, you can write a script to download the HTML content and save it in a separate folder on your file system. Then, run the Tailwind script to parse these files and generate the corresponding declarations in CSS.

2. If the content is dynamic, you can either put the classes you use as a comment in any of the files or set up a [safelist](https://tailwindcss.com/docs/content-configuration#safelisting-classes) in the Tailwind Config.

3. You might consider using the [@tailwindcss/typography](https://tailwindcss.com/docs/typography-plugin) plugin to style the content. Or, use an HTML parser to parse the HTML content and render it as Vue elements, where you can apply your custom styles.

4. For more complex cases, think about using plain CSS. Tailwind CSS is not always the best solution, and plain CSS might give you a better experience when working with dynamic content.
